### ISSUE TITLE
IDEA editor wrongly report error about configure ExtraPropertiesExtension

### ISSUE THREAD

**Issue by quaff** [gradle/gradle#24986]
### Expected Behavior

No error reports

```kotlin
ext {
	FileInputStream("$rootDir/another.properties").use {
		Properties().apply { load(it) }.forEach {
			val key = it.key as String
			if (!has(key))
				set(key, it.value)
		}
	}
}
```

### Current Behavior

![1](https://user-images.githubusercontent.com/143040/236149578-30619129-aea5-4f8a-b5a0-5ff4699c660f.jpg)

### Steps to Reproduce

Import [example.zip](https://github.com/gradle/gradle/files/11395305/example.zip) to latest IDEA CE.

```
IntelliJ IDEA 2023.1 (Community Edition)
Build #IC-231.8109.175, built on March 28, 2023
Runtime version: 17.0.6+10-b829.5 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 13.3.1
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 12
Metal Rendering is ON
Registry:
    debugger.new.tool.window.layout=true
    ide.experimental.ui=true

Non-Bundled Plugins:
    spring-javaformat (0.0.28)
    org.asciidoctor.intellij.asciidoc (0.38.13)

Kotlin: 231-1.8.20-IJ8109.175
```

`gradle build` works fine.

### Gradle version

8.1.1

**Comment by eskatos** [comment]
Sorry for the late reply.

It looks to be fixed using Gradle 8.4 and IntellIJ 2023.2

![image](https://github.com/gradle/gradle/assets/132773/b2caa671-234f-4cf6-8f89-1c5be4ac24d4)

**Comment by quaff** [comment]
Yes, I fixed now.
